gsmphys/samfshalcnv.f

```diff
@@ -1187,9 +1226,18 @@ subroutine samfshalcnv(im,ix,km,delt,itc,ntc,ntk,ntr,delp,
               dz    = zi(i,k) - zi(i,k-1)
               tem  = 0.25 * bb1 * (drag(i,k)+drag(i,k-1)) * dz
               tem1 = 0.5 * bb2 * (buo(i,k)+buo(i,k-1)) * dz
-              ptem = (1. - tem) * wu2(i,k-1)
-              ptem1 = 1. + tem
-              wu2(i,k) = (ptem + tem1) / ptem1
+              ! kgao 12/18/2023 - considers shear effect on updraft
+              if (use_shear_conv) then
+                tem2 = wush(i,k) * sqrt(wu2(i,k-1))
+                tem2 = (tem1 - tem2) * dz
+                ptem = (1. - tem) * wu2(i,k-1)
+                ptem1 = 1. + tem
+                wu2(i,k) = (ptem + tem2) / ptem1
+              else
+                ptem = (1. - tem) * wu2(i,k-1)
+                ptem1 = 1. + tem
+                wu2(i,k) = (ptem + tem1) / ptem1
+              endif
```

Comment:
ptem and ptem1 are based only on tem and wu2 and could be left outside the if-test.